fn/option.go

```diff
@@ -56,9 +54,22 @@ func (o Option[A]) UnwrapOrFunc(f func() A) A {
 	return ElimOption(o, f, func(a A) A { return a })
 }
 
+// Testing is a type passed to Test functions to manage test state. It has a
+// subset of testing.T methods needed by Option.UnwrapOrFail and
+// Result.UnwrapOrFail.
+type Testing interface {
+	// Helper marks the calling function as a test helper function.
+	Helper()
+
+	// Fatalf formats its arguments according to the format, analogous to
+	// Printf, and records the text in the error log, then marks the
+	// function as having failed and stops its execution.
+	Fatalf(format string, args ...any)
+}
+
```

Comment:
Do not put this in option.go

Reply:
Also I'm fairly certain that you can just index into *lntest.HarnessTest.T and just pass that.

Reply:
Where is a good place for the interface? It is used in option.go and result.go.

Yes, it is possible to pass `*lntest.HarnessTest.T`. I changed this method for consistency with `require` package. It is possible to pass `*lntest.HarnessTest` to `require`s functions and this feature is used in the codebase. The code looks cleaner if all checks pass `ht`, not `ht` and `ht.T` in different cases.

Reply:
Right but the cost of that is that you're placing completely unrelated and likely underdefined test interfaces in modules that are extremely primitive like option/result. At a bare minimum this testing interface should go in its own file in the fn package but even then I'd recommend avoiding that. Just pass ht.T at the call sites.

In order for interfaces to be well-maintained you need to have good definitions and behavioral properties for them. This one *seems* very ad hoc. If it *isn't* ad hoc, then it should be defined in its own canonical place and should have a good description of what the interface is fundamentally responsible for.

Reply:
I removed the commit. It is unrelated to the PR.

There are more cases of T-like types, e.g.  *B (benchmark) or `*assert.CollectT` (inside [require.EventuallyWithT](https://pkg.go.dev/github.com/stretchr/testify/require#EventuallyWithT)). My approach covers *B, but does not cover `*assert.CollectT`, so my commit needs reworking, maybe to change the subset of exposed methods.